tests/conftest.py

```diff
@@ -97,3 +97,36 @@ def adata_clustered(adata_neighbors):
         adata_obj, label_key="celltype", cluster_key="cluster", verbose=True
     )
     yield adata_obj
+
+
+DATASETS = {
+    "c_elegans": "https://github.com/Munfred/wormcells-data/releases/download/cao2017/cao2017.h5ad",
+    "zebrafish": "https://figshare.com/ndownloader/files/27265280",
+    # from https://cellrank.readthedocs.io/en/stable/_modules/cellrank/datasets.html
+}
+
+
+@pytest.fixture()
+def adata_from_url(request):
+    from io import BytesIO
+
+    import requests
+
+    dataset_name = request.param
+    url = DATASETS[dataset_name]
+
+    response = requests.get(url)
+    response.raise_for_status()
+
+    adata = sc.read_h5ad(BytesIO(response.content))
```

Comment:
scanpy can read URLs natively:
```suggestion
    adata = sc.read("placeholder_path.h5ad", backup_url=url)
```

The file will be saved to `placeholder_path` this way, so maybe worth creating a temp file

Reply:
Thanks, I totally forgot about that feature!